components/remote_settings/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+pub mod error;
+use config::RemoteSettingsConfig;
+pub use error::{RemoteSettingsError, Result};
+mod config;
+use rs_client::{Client, ClientConfig};
+use serde::Deserialize;
+use std::{fs::File, io::prelude::Write};
+
+uniffi_macros::include_scaffolding!("remotesettings");
+
+pub struct RemoteSettings {
+    pub config: RemoteSettingsConfig,
+    client: Client,
+}
+
+impl RemoteSettings {
+    pub fn new(config: RemoteSettingsConfig) -> Result<Self> {
+        Ok(RemoteSettings {
+            config: config.clone(),
+            client: Client::new(config.into())?,
+        })
+    }
+
+    pub fn get_records(&self) -> Result<Vec<JsonObject>> {
+        let resp = self.client.get_records()?;
+        Ok(resp.json::<RemoteSettingsResponse<_>>()?.data)
+    }
+
+    pub fn download_attachment_to_path(
+        &self,
+        attachment_location: String,
+        path: String,
+    ) -> Result<()> {
+        let resp = self.client.get_attachment(&attachment_location)?;
+        let mut file = File::create(path)?;
+        file.write_all(&resp.body)?;
+        Ok(())
+    }
+
+    pub fn get_records_since(&self, last_modified: i64) -> Result<Vec<JsonObject>> {
+        let resp = self.client.get_records_since(last_modified)?;
+        Ok(resp.json::<RemoteSettingsResponse<_>>()?.data)
+    }
+}
+
+impl From<RemoteSettingsConfig> for ClientConfig {
+    fn from(c: RemoteSettingsConfig) -> ClientConfig {
+        ClientConfig {
+            server_url: c.server_url,
+            bucket_name: c.bucket_name,
+            collection_name: c.collection_name,
+        }
+    }
+}
+
+#[derive(Deserialize)]
+struct RemoteSettingsResponse<T> {
+    data: Vec<T>,
+}
+
+#[derive(Deserialize)]
```

Comment:
Why the custom `JsonObject` instead of using `serde_json::Value` directly? IOW, something like `use serde_json::Value as JsonObject;` at the top of the file will work fine with a few other adjustments I've I've noted: